src/libcharon/plugins/kernel_super/kernel_super_ipsec.c

```diff
@@ -756,7 +778,32 @@ static status_t add_policy_internal(private_kernel_super_ipsec_t* this,
 				}
 
 				*should_add_routes = true;
+
+				if (IKE_SUPER_PEER_INFO_TYPE_TUNNEL == super_peer_info->get_type(super_peer_info)) {
+					break;
+				}
+				const char *vrf_name = super_peer_info->get_vrf_name(super_peer_info);
+				const char *device_id = super_peer_info->get_device_id(super_peer_info);
 				write_tunnel_oper_items(this, data, id, super_peer_info, data->child_sa, true);
+				if (! oper_db_writer_set_uptime(vrf_name,
+										  device_id,
+										  LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_LIFETIME_IKE_SA_UPTIME)) {
+					// Reason already logged @ function
+					break;
+				}
+				if (! oper_db_writer_set_uptime(vrf_name,
+										  device_id,
+										  LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_LIFETIME_IKE_SA_AUTH_UPTIME)) {
```

Comment:
Both `IKE_SA_UPTIME` and `IKE_SA_AUTH_UPTIME` shouldn't be set here. kernel-super only handles IPsec SAs.
In particular it means that on break-before-make child rekey the IKE uptime lifetime will be set here, even though it shouldn't.